library/src/main/java/com/novoda/downloadmanager/WaitForDownloadServiceThenPerform.java

```diff
@@ -0,0 +1,52 @@
+package com.novoda.downloadmanager;
+
+import android.support.annotation.Nullable;
+
+import com.novoda.notils.logger.simple.Log;
+
+final class WaitForDownloadServiceThenPerform {
+
+    interface Action<T> {
```

Comment:
I don't think we need this - we can use Java 8's `Supplier` functional interface

Reply:
Hang on yeah we do, proper Java 8 support requires API 24 :(

Reply:
`Supplier` requires level 24 😬